src/lib_json/json_value.cpp

```diff
@@ -1018,6 +1018,34 @@ Value Value::removeMember(const std::string& key) {
   return removeMember(key.c_str());
 }
 
+bool Value::removeIndex(ArrayIndex index, Value* removed) {
+  if (this->type_ != arrayValue) {
```

Comment:
Is the 'this->' notation really necessary? I find it distracting when accessing data members.
There are some corner cases where you need it, and so seeing it unnecessarily suggests to me that we're in one of those corner cases, and I waste a little attention on it. It's also just making the code a little bloaty, IMO.